server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailHeader.scala

```diff
@@ -75,12 +76,35 @@ object GroupedAddressesHeaderValue extends EmailHeaderValue {
   }
 }
 
+object MessageIdsHeaderValue {
+  def from(field: Field): MessageIdsHeaderValue = {
+    val messageIds: List[HeaderMessageId] = format(field.getBody)
+      .split(" ")
+      .flatMap(body => {
+        if(Pattern.compile("^<.*@.*>$").matcher(body).matches()) {
+          scala.Right(body.replaceAll("[<,>]", ""))
+        } else {
+          Left()
```

Comment:
This fallback is not good enough...

Reply:
We need a log message or sth?

Reply:
IMO `Right(body)`

Prefer returning potentially invalid elements other dropping some others.

(that's at least what I tested for messageId related convenience header)